src/common/types/core.js

```diff
@@ -1,5 +1,77 @@
 // @flow
 import { ITEM_TYPE_FOLDER, ITEM_TYPE_FILE, ITEM_TYPE_WEBLINK } from '../constants';
+import {
+    ACCESS_OPEN,
+    ACCESS_COLLAB,
+    ACCESS_COMPANY,
+    VIEW_SEARCH,
+    VIEW_FOLDER,
+    VIEW_ERROR,
+    VIEW_SELECTED,
+    VIEW_RECENTS,
+    VIEW_UPLOAD_EMPTY,
+    VIEW_UPLOAD_IN_PROGRESS,
+    VIEW_UPLOAD_SUCCESS,
+    VIEW_METADATA,
+    SORT_ASC,
+    SORT_DESC,
+    DELIMITER_SLASH,
+    DELIMITER_CARET,
+    SIZE_SMALL,
+    SIZE_MEDIUM,
+    SIZE_LARGE,
+    SIZE_VERY_LARGE,
+    FIELD_DATE,
+    FIELD_NAME,
+    FIELD_SIZE,
+    FIELD_RELEVANCE,
+    DEFAULT_VIEW_RECENTS,
+    DEFAULT_VIEW_FILES,
+    VERSION_RETENTION_DELETE_ACTION,
+    VERSION_RETENTION_REMOVE_ACTION,
+    VERSION_RETENTION_INDEFINITE,
+} from '../../constants';
```

Comment:
Should type files import source files? Or should the types for these constants live in their own type file and be imported into the source files?

Reply:
Most of these constants are used elsewhere besides these types... I'm inclined to have it as is personally.

I was kind of thinking to try to combine `src/common/constants.js` with `src/constants.js` at some point in a followup though. Seems redundant to me.

Reply:
most of the constants are defined like `const X: 'x': 'x'` so they can be used with `typeof X` in types and `X` in source files and get strict constant values

Reply:
I like the inline type declarations for constants. The main concern I have with this change is the possible circular dependencies it could introduce. @tjgupta mentioned he already ran into such an issue during his work on this PR. If that's not something worth worrying about further, though, then feel free to disregard.

Reply:
Yep it caused only one issue, and that was here:
https://github.com/box/box-ui-elements/pull/1775/files#diff-a7748843fbda5167d065231e3a2e7a1fL375

Arguably though, that isn't really a constant in my opinion. Probably doesn't really belong in that file.